recipes/libnova/all/patches/0004-constellation-syntax-error.patch

```diff
@@ -0,0 +1,20 @@
+--- a/src/libnova/constellation.h
++++ b/src/libnova/constellation.h
+@@ -19,6 +19,8 @@
+ #ifndef _LN_CONSTELLATION_H
+ #define _LN_CONSTELLATION_H
+ 
++#include <libnova/ln_types.h>
++
+ #ifdef __cplusplus
+ extern "C" {
+ #endif
+@@ -31,7 +33,7 @@ extern "C" {
+ * \ingroup constellations
+ * \brief Returns name of the constellation at the given position
+ */
+-const char* LIBNOVA_EXPORT ln_get_constellation(struct ln_equ_posn *position);
++const char LIBNOVA_EXPORT *ln_get_constellation(struct ln_equ_posn *position);
```

Comment:
This patch is weird.
The `*` should remain after the `const char`.
What compiler error'ed here with what error?

Reply:
Visual Studio shared:

```
FAILED: source_subfolder/src/CMakeFiles/nova.dir/utility.c.obj
C:\PROGRA~2\MICROS~2\2019\COMMUN~1\VC\Tools\MSVC\1429~1.300\bin\Hostx64\x64\cl.exe  /nologo -DHAVE_ROUND -DLIBNOVA_SHARED -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_USE_MATH_DEFINES -D__WIN32__ -Dnova_EXPORTS -Isource_subfolder\src /DWIN32 /D_WINDOWS /W3  /MP8 /MD /O2 /Ob2 /DNDEBUG /showIncludes /Fosource_subfolder\src\CMakeFiles\nova.dir\utility.c.obj /Fdsource_subfolder\src\CMakeFiles\nova.dir\ /FS -c source_subfolder\src\utility.c
source_subfolder\src\libnova/constellation.h(34): error C2059: syntax error: 'type'
```

the syntax is weird yes, actually it should be:

```c++
LIBNOVA_EXPORT const char *ln_get_constellation(struct ln_equ_posn *position);
```

Reply:
New c++ flamewar: east macro vs west macro 😄